compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3786,15 +3793,14 @@ private boolean isReferencingNonWorker(BLangExpression expression) {
             BLangSimpleVarRef simpleVarRef = (BLangSimpleVarRef) expression;
             BSymbol varRefSymbol = simpleVarRef.symbol;
             if (varRefSymbol == null) {
-                return true; // TODO: inconclusive
+                return null; // TODO: inconclusive
             }
             String varRefSymbolName = varRefSymbol.getName().value;
             if (workerExists(env, varRefSymbolName)) {
-                return false;
+                return varRefSymbolName;
             }
-            nonWorkerReferenceName = varRefSymbolName;
-        } 
-        return true;
+        }
+        return "";
```

Comment:
what does empty string represent here?

Reply:
I will refactor the code. 
As of now if no worker is found for the name it will return "".